### PR TITLE
Module 4 exercise

### PR DESCRIPTION
**Pull request by lukestern**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+sudo apt-get update && sudo apt-get -y upgrade
+
+echo 'Installing pyenv and python dependancies.'
+sudo apt-get install -y --no-install-recommends build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git make build-essential libxmlsec1-dev libxml2-dev libsqlite3-dev
+
+echo 'Cloning pyenv repo.'
+git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+cd ~/.pyenv && src/configure && make -C src
+
+echo 'Difining environmant variable PYENV_ROOT'
```
Comment (BenRamchandani) [inline Vagrantfile]: Typos

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
+$script = <<-SCRIPT
+echo 'Updating package repos and updating packages.'
+sudo apt-get update && sudo apt-get -y upgrade
+
+echo 'Installing pyenv and python dependancies.'
```
Comment (BenRamchandani) [inline Vagrantfile]: dependencies

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+echo 'Cloning pyenv repo.'
+git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+cd ~/.pyenv && src/configure && make -C src
+
+echo 'Difining environmant variable PYENV_ROOT'
+echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bash_profile
+echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bash_profile
+
+echo 'Add pyenv init to shell'
+echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi' >> ~/.bash_profile
```
Comment (BenRamchandani) [inline Vagrantfile]: You'll need to change this to `pyenv init --path` or add `echo 'eval "$(pyenv init --path)"' >> ~/.bash_profile` above to get it working - I think this is `pyenv` changing under us.

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+echo 'Installing pyenv and python dependancies.'
+sudo apt-get install -y --no-install-recommends build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git make build-essential libxmlsec1-dev libxml2-dev libsqlite3-dev
+
+echo 'Cloning pyenv repo.'
+git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+cd ~/.pyenv && src/configure && make -C src
+
+echo 'Difining environmant variable PYENV_ROOT'
+echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bash_profile
```
Comment (BenRamchandani) [inline Vagrantfile:13]: Mostly as a point of interest creating a `.bash_profile` file will stop bash reading `.profile`, I don't think it matters here.

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+SCRIPT
+
+$app_setup = <<-SCRIPT
+cd /vagrant
+
+echo 'Installing TO-DO App dependencies.'
+poetry install
+
+echo ' Launching App.'
+sudo -u vagrant poetry run gunicorn -b 0.0.0.0:5000 'todo_app.app:create_app()'
```
Comment (BenRamchandani) [inline Vagrantfile]: gunicorn has a `--daemon` flag which will stop it blocking `vagrant up`

File: Vagrantfile (hunk 1 above)
Comment (BenRamchandani) [reply Vagrantfile]: I use the `streetsidesoftware.code-spell-checker` VSCode extension for spell checking and find it works pretty well.

**Review by BenRamchandani — APPROVED** [review]
This needed a tweak to get it working, but I think that's `pyenv` changing under us so passing anyway.

If you can fix it that's great, but if not that's fine, as you know we switch to Docker from the next exercise anyway.

File: Vagrantfile (hunk 5 above)
Comment (lukestern) [reply Vagrantfile]: Change made

File: Vagrantfile (hunk 3 above)
Comment (lukestern) [reply Vagrantfile]: Change made

**Pull request merged**